pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -184,13 +185,18 @@ func BuildListenerTLSContext(serverTLSSettings *networking.ServerTLSSettings,
 	}
 
 	// Set TLS parameters if they are non-default
-	if len(serverTLSSettings.CipherSuites) > 0 ||
+	var ecdhCurves []string
+	if mesh != nil && mesh.MeshExternal_TLS != nil {
+		ecdhCurves = mesh.MeshExternal_TLS.EcdhCurves
+	}
+	if len(serverTLSSettings.CipherSuites) > 0 || len(ecdhCurves) > 0 ||
 		serverTLSSettings.MinProtocolVersion != networking.ServerTLSSettings_TLS_AUTO ||
 		serverTLSSettings.MaxProtocolVersion != networking.ServerTLSSettings_TLS_AUTO {
 		ctx.CommonTlsContext.TlsParams = &auth.TlsParameters{
 			TlsMinimumProtocolVersion: convertTLSProtocol(serverTLSSettings.MinProtocolVersion),
 			TlsMaximumProtocolVersion: convertTLSProtocol(serverTLSSettings.MaxProtocolVersion),
 			CipherSuites:              serverTLSSettings.CipherSuites,
+			EcdhCurves:                ecdhCurves,
```

Comment:
I think this has to be set only if len(EcdhCurves) > 0. Otheriwse it will overwrite the Envoys default Ecdh curves

Reply:
I tested this out and this seems to be working as expected.